web/app/js/components/MetricsTable.jsx

```diff
@@ -37,97 +108,60 @@ const columnDefinitions = (resource, showNamespaceColumn, PrefixedLink) => {
-      sorter: (a, b) => numericSort(a.P50, b.P50)
-    },
-    {
-      title: "P95 Latency",
-      dataIndex: "P95",
-      isNumeric: true,
-      render: d => metricToFormatter["LATENCY"](d.P95),
-      sorter: (a, b) => numericSort(a.P95, b.P95)
-    },
-    {
-      title: "P99 Latency",
-      dataIndex: "P99",
-      isNumeric: true,
-      render: d => metricToFormatter["LATENCY"](d.P99),
-      sorter: (a, b) => numericSort(a.P99, b.P99)
-    },
-    {
-      title: "TLS",
-      dataIndex: "tlsRequestPercent",
-      isNumeric: true,
-      render: d => _isNil(d.tlsRequestPercent) || d.tlsRequestPercent.get() === -1 ? "---" : d.tlsRequestPercent.prettyRate(),
-      sorter: (a, b) => numericSort(
-        a.tlsRequestPercent ? a.tlsRequestPercent.get() : -1,
-        b.tlsRequestPercent ? b.tlsRequestPercent.get() : -1)
-    },
-    {
-      title: "Grafana",
-      key: "grafanaDashboard",
-      isNumeric: true,
-      render: row => {
-        if (!isAuthorityTable && (!row.added || _get(row, "pods.totalPods") === "0") ) {
-          return null;
-        }
-
-        return (
-          <GrafanaLink
-            name={row.name}
-            namespace={row.namespace}
-            resource={row.type}
-            PrefixedLink={PrefixedLink} />
-        );
+  let grafanaColumn = {
+    title: "Grafana",
+    key: "grafanaDashboard",
+    isNumeric: true,
+    render: row => {
+      if (!isAuthorityTable && (!row.added || _get(row, "pods.totalPods") === "0") ) {
+        return null;
       }
+
+      return (
+        <GrafanaLink
+          name={row.name}
+          namespace={row.namespace}
+          resource={row.type}
+          PrefixedLink={PrefixedLink} />
+      );
     }
-  ];
+  };
+
+  let nameColumn = {
+    title: isMultiResourceTable ? "Resource" : friendlyTitle(resource).singular,
+    dataIndex: "name",
+    isNumeric: false,
+    render: d => {
+      let nameContents;
+      if (resource === "namespace") {
+        nameContents = <PrefixedLink to={"/namespaces/" + d.name}>{d.name}</PrefixedLink>;
+      } else if (!d.added || isAuthorityTable) {
+        nameContents = getResourceDisplayName(d);
+      } else {
+        nameContents = (
+          <PrefixedLink to={"/namespaces/" + d.namespace + "/" + d.type + "s/" + d.name}>
+            {getResourceDisplayName(d)}
+          </PrefixedLink>
+        );
+      }
+      return (
+        <Grid container alignItems="center" spacing={8}>
+          <Grid item>{nameContents}</Grid>
+          { _isEmpty(d.errors) ? null :
+          <Grid item><ErrorModal errors={d.errors} resourceName={d.name} resourceType={d.type} /></Grid>}
+        </Grid>
+      );
+    },
+    sorter: (a, b) => (getResourceDisplayName(a) || "").localeCompare(getResourceDisplayName(b))
+  };
+
+  let columns = [nameColumn];
+  if (isTcpTable) {
+    columns = columns.concat(TcpStatColumns);
+  } else {
+    columns = columns.concat(httpStatColumns);
+    columns = columns.concat(grafanaColumn);
```

Comment:
i just realized while testing this that we don't expose tcp-level stats on the `Linkerd Pod` Grafana dashboard. if/when we add tcp metrics to that dashboard, perhaps we should add the Grafana column when `isTcpTable == true`, what do you think?

Reply:
true! filed https://github.com/linkerd/linkerd2/issues/2329